src/common/health.rs

```diff
@@ -0,0 +1,104 @@
+use std::collections::HashSet;
+use std::ops::DerefMut as _;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use collection::shards::replica_set::ReplicaState;
+use collection::shards::shard::ShardId;
+use collection::shards::CollectionId;
+use storage::content_manager::toc::TableOfContent;
+use tokio::sync::RwLock;
+
+pub struct Ready {
+    toc: Arc<TableOfContent>,
+    consensus_sync_instant: Instant,
+    initial_unhelathy_shards: RwLock<Option<HashSet<Replica>>>,
+}
+
+impl Ready {
+    pub fn new(toc: Arc<TableOfContent>) -> Self {
+        Self::with_consensus_sync_timeout(toc, Duration::from_secs(10))
```

Comment:
Instead of using `10` directly, maybe we want to use the `CONSENSUS_META_OP_WAIT` constant.